### ISSUE TITLE
UDP version of RoundTrip() available?

### ISSUE THREAD

**Issue by WinstonPrivacy** [xtaci/kcptun#608]
I'm experimenting with using kcptun as a drop-in replacement for Golang's native TCP connection stack. It mostly works however, it is not a drop-in replacement. A big problem is that http/transport.go/RoundTrip() frequently returns EOF errors when making TLS requests. This appears to be dependent on timing, for instance a single request never returns EOF, but a batch of 25 requests will often result in the first few requests succeeding and the rest erroring out.

Tracing this down, it appears that readloop() within the native RoundTrip() method is checking for server disconnects and this is causing the problem. The logic is convoluted and difficult to understand however.

Turning off KeepAlives in the transport helps a great deal but doesn't completely cure the problem.

Is anyone aware of a replacement for transport (or RoundTrip) that could work with kcptun?

**Comment by xtaci** [comment]
have u solved this ?

**Comment by WinstonPrivacy** [comment]
Yes - we had to turn off keepalives, add an idle timeout to smux/stream.go so that streams would close after 5 seconds of inactivity and set a read timeout on the receiving end during the handshake. This fixed most of our issues though we're still working on some odd edge cases.

**Comment by Immueggpain** [comment]
will smux proactively drop streams?
